.github/workflows/My-ci-pipeline.yml

```
@@ -28,4 +28,79 @@ jobs:
       run: docker build --target test --tag todo-app:test .
     
     - name: BuildAndTestImage
-      run: docker run --env-file .env.test todo-app:test tests
\ No newline at end of file
+      run: docker run --env-file .env.test todo-app:test tests
+
+  Publish:
+    if: ${{ github.ref == 'refs/heads/main' && github.event_name == 'push' }}
+    #if: ${{ github.ref == 'refs/heads/module8' && github.event_name == 'push' }}
+    name: Publish
+    runs-on: ubuntu-latest
+    needs: [build]
+    steps:
+    - uses: actions/checkout@v2
+    
+    - uses: satackey/action-docker-layer-caching@v0.0.11
+      continue-on-error: true
+
+    - name: publish to docker hub
+      run: echo "Publishing!"
+
+    #- name: Build Image
+    #  run: docker build --tag latest .
+    - name: Build Image
+      run: docker build --target production --tag moudule8 .
+
+    - name: Change Tag
+      run: docker tag moudule8 wishy78/module8:module8
```

Why do you build it with one tag then immediately swap tags to push it? There's no harm, but it feels unnecessary, we could instead just:
```suggestion
      run: docker build --target production --tag wishy78/module8:module8 .
```